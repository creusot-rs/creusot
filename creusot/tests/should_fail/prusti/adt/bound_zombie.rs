extern crate creusot_contracts;
use creusot_contracts::prusti_prelude::*;

trait Trait {}

impl<T> Trait for T {}

pub struct Wrap<T: Trait>(T);

#[open]
#[logic('x where 'curr: 'x)]
pub fn test_constructor<'x>(x: Box<u32>) -> Wrap<Box<u32>> {
    Wrap(at_expiry::<'x>(x))
}