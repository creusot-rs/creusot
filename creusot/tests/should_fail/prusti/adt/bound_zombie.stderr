error[creusot]: expected `Wrap<std::boxed::Box<u32>>` found `Wrap<creusot_contracts::__stubs::__prusti::Zombie<std::boxed::Box<u32>>>`
  --> bound_zombie.rs:13:5
   |
13 |     Wrap(at_expiry::<'x>(x))
   |     ^^^^^^^^^^^^^^^^^^^^^^^^

error: aborting due to previous error

